thoth-api/src/model/mod.rs

```diff
@@ -542,6 +560,84 @@ macro_rules! db_insert {
     };
 }
 
+/// Declares a change ordinal function implementation for any insertable which
+/// has an ordinal field. Useful together with the `Reorder` trait.
+///
+/// Example usage
+/// -------------
+///
+/// ```ignore
+/// use crate::db_change_ordinal;
+/// use crate::model::Reorder;
+/// use crate::schema::contribution;
+///
+/// impl Reorder for Contribution {
+///     db_change_ordinal!(
+///         contribution::table,
+///         contribution::contribution_ordinal,
+///         "contribution_contribution_ordinal_work_id_uniq",
+///     );
+/// }
+/// ```
+///
+///
+#[cfg(feature = "backend")]
+#[macro_export]
+macro_rules! db_change_ordinal {
+    ($table_dsl:expr,
+     $ordinal_field:expr,
+     $constraint_name:literal) => {
+        fn change_ordinal(
+            &self,
+            db: &$crate::db::PgPool,
+            current_ordinal: i32,
+            new_ordinal: i32,
+            account_id: &Uuid,
+        ) -> ThothResult<Self> {
+            let other_objects = self.get_other_objects(db)?;
+            // Execute all updates within the same transaction,
+            // because if one fails, the others need to be reverted.
+            let mut connection = db.get()?;
+            connection.transaction(|connection| {
+                if current_ordinal == new_ordinal {
+                    // No change required. Caller should have checked this.
+                    unreachable!()
```

Comment:
let's avoid `unreachable`, as technically this is reachable – just return `Ok(self.clone());` in that case. Also adding `Sized + Clone` to the `Reorder` trait